megamek/src/megamek/common/LosEffects.java

```diff
@@ -359,31 +359,41 @@ public static LosEffects calculateLOS(final IGame game, final @Nullable Entity a
      * terrain between the attacker and target. Checks to see if the attacker
      * and target are at an angle where the LOS line will pass between two
      * hexes. If so, calls losDivided, otherwise calls losStraight.
+     *
+     * @param game the game to calculate using
+     * @param attacker the attacker, which may be null. If it is, the view is blocked.
+     * @param target the target, which may be null. If it is, the view is blocked.
+     * @param spotting if the person is spotting
+     * @return the found LOS Effects
      */
     public static LosEffects calculateLOS(final IGame game, final @Nullable Entity attacker,
                                           final @Nullable Targetable target, final boolean spotting) {
-        // we need an extra step here, because units with secondary position can calculate LoS
-        // from hexes other than that returned from getPosition()
-        // create a vector of attacker position and a vector of target positions - double loop
-        // through them both and select the best one
-        Vector<Coords> attackerPositions = new Vector<>();
-        attackerPositions.add(attacker.getPosition());
+        if ((attacker == null) || (target == null)) {
+            return calculateLOS(game, attacker, target,
+                    (attacker == null) ? null : attacker.getPosition(),
+                    (target == null) ? null : target.getPosition(), spotting);
+        }
+
+        // We need to create Attacker and Target position lists because they might have secondary
+        // positions that would be better
+        final List<Coords> attackerPositions = new ArrayList<>();
         // if a grounded DropShip is the attacker, then it gets to choose the best secondary position for LoS
         if ((attacker instanceof Dropship) && !attacker.getSecondaryPositions().isEmpty()) {
```

Comment:
Maybe we don't really care if the attacker is a dropship? I can see this causing bugs in the future for other units that have secondary positions.

Reply:
Not really part of this PR, just a thought that occurred to me as I was reading it.

Reply:
I wondered about it, but just assumed it was okay. Looked into it, I believe we can safely remove the check.